### PR TITLE
[16.0][MIG] sale_payment_sheet: Migration to version 16.0

### PR DESCRIPTION
**Pull request by carolinafernandez-tecnativa**
-Fixes PR Standard migration to v16
-Improve Tests
-Supersedes  https://github.com/OCA/sale-workflow/pull/2554 due to changes requested not attended.

@Tecnativa

TT44357

@pedrobaeza

### CODE REVIEW

**Comment by carolinafernandez-tecnativa** [comment]
ping @pedrobaeza

File: sale_payment_sheet/models/sale_payment_sheet.py [hunk 1]
```diff
@@ -183,7 +182,6 @@ def button_reopen(self):
             raise UserError(
                 _("You can not reopen a sheet that has any reconciled line.")
             )
-        self_sudo.statement_id.button_reopen()
```
Comment (pedrobaeza) [inline sale_payment_sheet/models/sale_payment_sheet.py]: Why this change?

File: sale_payment_sheet/models/sale_payment_sheet.py [hunk 2]
```diff
@@ -136,8 +136,6 @@ def button_confirm_sheet(self):
                 {
                     "name": sheet.name,
                     "date": sheet.date,
-                    "journal_id": sheet.journal_id.id,
-                    "user_id": sheet.user_id.id,
```
Comment (pedrobaeza) [inline sale_payment_sheet/models/sale_payment_sheet.py]: Why this change?

File: sale_payment_sheet/models/sale_payment_sheet.py (hunk 1 above)
Comment (carolinafernandez-tecnativa) [reply sale_payment_sheet/models/sale_payment_sheet.py]: button_reopen() does not exist anymore. I add unlink statement as it is on v15.

File: sale_payment_sheet/models/sale_payment_sheet.py (hunk 2 above)
Comment (pedrobaeza) [reply sale_payment_sheet/models/sale_payment_sheet.py]: And the salesman?

Comment (carolinafernandez-tecnativa) [reply sale_payment_sheet/models/sale_payment_sheet.py]: I add journal_id again, user_id does not exist anymore in v16.
I did not realize previous PR delete this two lines.

Comment (pedrobaeza) [reply sale_payment_sheet/models/sale_payment_sheet.py]: OK. It seems the journal is added later in this method. Any reason?

Comment (carolinafernandez-tecnativa) [reply sale_payment_sheet/models/sale_payment_sheet.py]: Journal must be added later(in account.bank.statement.line) because journal_id in the account.bank.statement model it is a computed field (takes journal_id from lines)

I added salesman too.

**Review by pedrobaeza — APPROVED** [review]

**Comment by sergio-teruel** [comment]
@carolinafernandez-tecnativa Can you review the commits history?
for example this one is missing..
https://github.com/OCA/sale-workflow/pull/2563/commits/87e9283178fe5f53fc949d6939bac191f8e85d7b

**Review by sergio-teruel — CHANGES_REQUESTED** [review]
Missing commits from older version

**Comment by carolinafernandez-tecnativa** [comment]
> @carolinafernandez-tecnativa Can you review the commits history? for example this one is missing.. [87e9283](https://github.com/OCA/sale-workflow/commit/87e9283178fe5f53fc949d6939bac191f8e85d7b)

@sergio-teruel done, can you please check it now? Thanks!

**Comment by sergio-teruel** [comment]
Can you review please...
image

**Comment by carolinafernandez-tecnativa** [comment]
> Can you review please... image

Done!  
image

File: sale_payment_sheet/wizards/sale_invoice_payment_view.xml [hunk 3]
```diff
@@ -19,7 +19,11 @@
 
                 </group>
                 <group>
-                    <field name="wiz_line_ids" nolabel="1">
+                    <field
+                        name="wiz_line_ids"
+                        nolabel="1"
+                        style="width:900px !important"
```
Comment (sergio-teruel) [inline sale_payment_sheet/wizards/sale_invoice_payment_view.xml]: Remove this style.

**Review by sergio-teruel — CHANGES_REQUESTED** [review]

File: sale_payment_sheet/wizards/sale_invoice_payment_view.xml (hunk 3 above)
Comment (carolinafernandez-tecnativa) [reply sale_payment_sheet/wizards/sale_invoice_payment_view.xml]: Done, fix view with colspan="2"

**Comment by carolinafernandez-tecnativa** [comment]
ping @pedrobaeza @sergio-teruel

**Review by sergio-teruel — APPROVED** [review]
